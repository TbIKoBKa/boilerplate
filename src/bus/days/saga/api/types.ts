// Types
import { Days } from '../../types';

export type FetchDaysContract = () => Promise<Days>
