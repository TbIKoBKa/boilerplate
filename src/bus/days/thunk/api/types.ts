// Types
import { Days } from '../../types';

export type FetchDays = () => Promise<Days>
