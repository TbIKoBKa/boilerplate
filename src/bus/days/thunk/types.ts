// Core
import { ActionReducerMapBuilder } from '@reduxjs/toolkit';

// Types
import * as types from '../types';

// Fetch
export const FETCH_WEATHER_ASYNC = 'days/FETCH_WEATHER_ASYNC';

// Contracts
export type FetchDaysAsyncContract = (builder: ActionReducerMapBuilder<types.Days>) => void;
