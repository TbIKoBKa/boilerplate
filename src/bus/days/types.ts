// Core
import { CaseReducer, PayloadAction } from '@reduxjs/toolkit';

// Types
export type WeatherType = 'sunny' | 'cloudy' | 'rainy'

// State
export type Day = {
    id: string
    rain_probability: number
    humidity: number
    day: number
    temperature: number
    type: WeatherType
}
export type Days = Day[]

// Contracts
export type SetWeatherContract = CaseReducer<Days, PayloadAction<Days>>
