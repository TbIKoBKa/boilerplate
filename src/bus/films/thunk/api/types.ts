// Types
import { Films } from '../../types';

export type FetchFilms = () => Promise<Films>
