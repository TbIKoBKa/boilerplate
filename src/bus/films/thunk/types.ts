// Fetch
export const FETCH_FILMS_ASYNC = 'FETCH_FILMS_ASYNC';
export type fetchFilmsActionAsync = {
    type: typeof FETCH_FILMS_ASYNC
}
export type FetchFilmsContract = () => fetchFilmsActionAsync
