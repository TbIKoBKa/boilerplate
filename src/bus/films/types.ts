// Core

export type Film = {
    title: string,
    episode_id: number,
    opening_crawl: string,
    director: string,
    producer: string,
    release_date: string,
    characters: Array<string>,
    planets: Array<string>,
    starships: Array<string>,
    vehicles: Array<string>,
    species: Array<string>,
    created: string,
    edited: string,
    url: string,
    [key: string]: any,
}

export type Films = Array<Film>;

// Set
export const SET_FILMS = 'SET_FILMS';
export type SetFilmsActionType = {
    type: typeof SET_FILMS,
    payload: Films,
}
export type SetFilmsContract = (payload: Films) => SetFilmsActionType

export type PeopleActionTypes =
    | SetFilmsActionType
