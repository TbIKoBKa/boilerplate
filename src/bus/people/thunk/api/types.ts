// Types
import { People } from '../../types';

export type FetchPeople = () => Promise<People>
