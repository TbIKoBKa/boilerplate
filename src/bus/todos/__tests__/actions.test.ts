// Actions
import {
    deleteTodoAction,
    setTodoAction,
    setTodosAction,
    updateTodoAction,
} from '../actions';
import {
    createTodoActionAsync,
    deleteTodoActionAsync,
    fetchTodosActionAsync,
    updateTodoActionAsync,
} from '../saga/actions';

// Types
import { Todos } from '../types';

const testTodos: Todos = [
    {
        id:          '0',
        isCompleted: false,
        text:        'text',
    },
];

describe('todos actions', () => {
    test('should delete todo', () => {
        expect(deleteTodoAction('0')).toMatchSnapshot();
    });
    test('should set todo', () => {
        expect(setTodoAction(testTodos[ 0 ])).toMatchSnapshot();
    });
    test('should set todos', () => {
        expect(setTodosAction(testTodos)).toMatchSnapshot();
    });
    test('should update todo', () => {
        expect(updateTodoAction(testTodos[ 0 ])).toMatchSnapshot();
    });
    test('should create todo async', () => {
        expect(createTodoActionAsync({ body: { text: 'text' }})).toMatchSnapshot();
    });
    test('should delete todo async', () => {
        expect(deleteTodoActionAsync({ todoId: '0' })).toMatchSnapshot();
    });
    test('should fetch todos async', () => {
        expect(fetchTodosActionAsync()).toMatchSnapshot();
    });
    test('should update todo async', () => {
        expect(updateTodoActionAsync({ todoId: '0', body: { isCompleted: true }})).toMatchSnapshot();
    });
});
