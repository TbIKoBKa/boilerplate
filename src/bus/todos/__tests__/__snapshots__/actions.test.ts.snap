// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todos actions should create todo async 1`] = `
Object {
  "payload": Object {
    "body": Object {
      "text": "text",
    },
  },
  "type": "CREATE_TODO_ASYNC",
}
`;

exports[`todos actions should delete todo 1`] = `
Object {
  "payload": "0",
  "type": "DELETE_TODO",
}
`;

exports[`todos actions should delete todo async 1`] = `
Object {
  "payload": Object {
    "todoId": "0",
  },
  "type": "DELETE_TODO_ASYNC",
}
`;

exports[`todos actions should fetch todos async 1`] = `
Object {
  "type": "FETCH_TODOS_ASYNC",
}
`;

exports[`todos actions should set todo 1`] = `
Object {
  "payload": Object {
    "id": "0",
    "isCompleted": false,
    "text": "text",
  },
  "type": "SET_TODO",
}
`;

exports[`todos actions should set todos 1`] = `
Object {
  "payload": Array [
    Object {
      "id": "0",
      "isCompleted": false,
      "text": "text",
    },
  ],
  "type": "SET_TODOS",
}
`;

exports[`todos actions should update todo 1`] = `
Object {
  "payload": Object {
    "id": "0",
    "isCompleted": false,
    "text": "text",
  },
  "type": "UPDATE_TODO",
}
`;

exports[`todos actions should update todo async 1`] = `
Object {
  "payload": Object {
    "body": Object {
      "isCompleted": true,
    },
    "todoId": "0",
  },
  "type": "UPDATE_TODO_ASYNC",
}
`;
